### ISSUE TITLE
Darknet Yolo v2 added to the OpenCV

### ISSUE THREAD

**Issue by AlexeyAB** [pjreddie/darknet#242]
Now Darknet Yolo v2 added to the OpenCV: https://github.com/opencv/opencv/pull/9705
You can use it from master-branch or since OpenCV 3.4.0 will be released.
There is example how to use Yolo v2: https://github.com/opencv/opencv/blob/master/samples/dnn/yolo_object_detection.cpp
For now it CPU-only (~10 times faster than original) and can be used on mobile devices, but in the future GPU will be supported.

If there will be many upvotes, perhaps I will add support for Yolo 9000 :)
**Don't write anything there, just upvote :)**

![yolo](https://user-images.githubusercontent.com/4096485/31386280-ea9db638-adce-11e7-9e07-e56ecd217340.jpg)

**Comment by MakerVisioneer** [comment]
video capture next, please?:)

**Comment by Myzhar** [comment]
@AlexeyAB I saw the issue in the OpenCV repository, but it's not clear if also the "GPU version" of YOLO will be available.

**Comment by bretzel59** [comment]
@AlexeyAB, I'm using your windows port of Darknet and I would like to use darknet with OpenCV. Do you think you can add GPU support ? And Yolo 9000 support of course :). It would be great !

Many Thanks for your valuable work !

**Comment by richipower** [comment]
@AlexeyAB actually i am using three versions of darknet, (yolo_v2 with opencv, yolo_v2 with cuda and yolo_v3 with cuda) 
now i am trying to get yolo_v3 working with opencv but the "shortcut" layer is not implemented, (i think is similar as the "route" layer from v2.. )  i dont know how i can put the "activation=linear" option on the opencv so i cannot read it, i was trying to create my own reader-function modifying the .src files but no success, do you have any ideas how this can be solved? or if there is any plan on adding support for yolo_v3 on opencv? 
thanks!

**Comment by deamonDevs** [comment]
@AlexeyAB I am writing code to use yolo_v2 with OpenCV in python but I am **not getting correct detections.**  can you please tell me how to parse the output detections. means how to get box coordinates and how to get class with class score.

Thanks in advance!

**Comment by abdulkalam1233** [comment]
@AlexeyAB  can  you explain about activation linear and activation leaky

**Comment by AlexeyAB** [comment]
Yolo v3 is added to the OpenCV too: https://github.com/opencv/opencv/pull/11322

**Comment by abdulkalam1233** [comment]
[net]
# Testing
batch=1
subdivisions=1
# Training
# batch=64
# subdivisions=2
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 500200
policy=steps
steps=400000,450000
scales=.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=512
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=425
activation=linear

[region]
anchors =  0.57273, 0.677385, 1.87446, 2.06253, 3.33843, 5.47434, 7.88282, 3.52778, 9.77052, 9.16828
bias_match=1
classes=80
coords=4
num=5
softmax=1
jitter=.2
rescore=0

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=1


why the activation is linear in last convolution layer? can you explain briefly?
[convolutional]
size=1
stride=1
pad=1
filters=425
activation=linear

**Comment by abdulkalam1233** [comment]
@AlexeyAB  Thanks man. You are awesome

**Comment by streamnsight** [comment]
is yolo9000 planned to be integrated? If not, what does it take to do this?

**Comment by AlexeyAB** [comment]
@streamnsight 
Yolo9000 isn't planned to be integrated (should be implemented softmax-tree in the `region_layer.cpp`)
May be better to use Yolov3 multilabel training instead of Yolo9000: https://arxiv.org/pdf/1804.02767v1.pdf

**Comment by streamnsight** [comment]
@AlexeyAB thanks for the note
I'm looking to use a trained model. Is there weight files available for more than 80 classes that work with the current OpenCV implementation?

**Comment by AlexeyAB** [comment]
@streamnsight 

`yolov3-openimages.cfg` is for 601 classes.

https://github.com/AlexeyAB/darknet#pre-trained-models-for-different-cfg-files-can-be-downloaded-from-smaller---faster--lower-quality

> `yolov3-openimages.cfg` (247 MB COCO Yolo v3) - requires 4 GB GPU-RAM: https://pjreddie.com/media/files/yolov3-openimages.weights

**Comment by streamnsight** [comment]
@AlexeyAB 
Thanks for the help. Sorry I'm kind new at this and having issues:

I got the `cfg` from https://github.com/AlexeyAB/darknet/blob/master/cfg/yolov3-openimages.cfg 
the class names from https://github.com/AlexeyAB/darknet/blob/master/data/openimages.names
and the weights from your link above https://pjreddie.com/media/files/yolov3-openimages.weights, but i get very poor results. I have to drop the confidence down to 5% to get anything and then results are very generic (like 'Animal' for a 'bear', or another where yolov3 gives me 'diningtable', 'bottle', 'wine glass', 'knife', returns just 'Tool')

Using the same exact code to load and eval the model. Am I missing something?

**Comment by pratikbhave2** [comment]
@AlexeyAB Can you also add support for yolo9000 please?

**Comment by kulkarnikeerti** [comment]
@AlexeyAB @pjreddie Can you please provide more information/support on yolo9000? How the data is prepared in a tree structure? How it is used for the model training?
